src/token/token.service.ts

```diff
@@ -1,22 +1,67 @@
-import { Injectable } from '@nestjs/common';
+import { HttpService, Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
 import { Message } from 'discord.js';
+import config from '../config';
+import { ValidationResponse } from './interfaces/token.interface';
 import { TokenCacheService } from './token-cache.service';
 
 @Injectable()
 export class TokenService {
-  constructor(private readonly tokenCacheService: TokenCacheService) {}
+  constructor(
+    private readonly tokenCacheService: TokenCacheService,
+    private readonly httpService: HttpService,
+    private readonly configService: ConfigService,
+  ) {}
 
   public async createToken(message: Message) {
     const serverId = message.guild.id;
+    let returnString: string;
     try {
       const token = await this.tokenCacheService.getNewToken(serverId, [
         'Data.Read',
       ]);
-      const returnString = `**New Token created**\n\nA new Accesstoken to access the Eddiehub-Api was created\n\nClientID 🆔:\n\`${token.clientId}\`\n\nAccessToken 🔑:\n\`${token.accessToken}\`\n\nScopes 🚀:\n\`${token.scopes}\``;
+      returnString = `**New Token created**\n\nA new Accesstoken to access the Eddiehub-Api was created\n\nClientID 🆔:\n\`${token.clientId}\`\n\nAccessToken 🔑:\n\`${token.accessToken}\`\n\nScopes 🚀:\n\`${token.scopes}\``;
       return returnString;
     } catch (error) {
-      const returnString = `**Token creation failed**\n\nThere was an error creating a new token`;
+      returnString = `**Token creation failed**\n\nThere was an error creating a new token`;
       return returnString;
     }
   }
+  public async validateToken(message: Message, token: string) {
+    const url = `${this.configService.get('API_URL')}/auth/validate`;
+    try {
+      const response = await this.httpService
+        .post(
+          url,
+          { token },
+          { headers: { 'Client-Token': this.configService.get('API_TOKEN') } },
+        )
+        .toPromise();
+      const body = response.data as ValidationResponse;
+      if (body.valid === false) {
+        return config
+          .defaultEmbed(config.colors.alerts)
+          .setTitle('Validation failed')
+          .setDescription('The Validation of your token was not successfull')
+          .setAuthor(
+            message.author.username,
+            message.author.displayAvatarURL(),
+          );
+      }
+      return config
```

Comment:
Validation Result is getting returned as an embed. I'm not quite sure about that. But when sending the request you need to pass the token in the command. This means the token gets public in every case.
The other option would be not just allow an approved client to validate a token

Reply:
oh good point 👍 could they:

1. run the command with no token, and get a DM back with instructions on how to run the validation command in the DM
2. then share the token for validation only via DM (new command)

if command is run in public with a token can the bot delete the message?

Reply:
The Command is now only executable when sent as a DM to the bot. In case its in public the bot replies that the user should send the command as a dm

Reply:
We should look in to slash commands for this.

Reply:
I need to learn more about slash commands to see pros/cons

Reply:
I dont know anything about them yet either 😦

Reply:
+1 for slash commands, I think the migration is still going on 🤔 Once it is done, we can implement slash commands